lib/Versions/VersionsBackend.php

```diff
@@ -332,9 +332,19 @@ public function deleteVersion(IVersion $version): void {
 	}
 
 	public function createVersionEntity(File $file): void {
+		$fileId = $file->getId();
+		$timestamp = $file->getMTime();
+		try {
+			$existingVersion = $this->groupVersionsMapper->findVersionForFileId($fileId, $timestamp);
+			if ($existingVersion instanceof GroupVersionEntity) {
+				return;
+			}
```

Comment:
```suggestion
			$this->groupVersionsMapper->findVersionForFileId($fileId, $timestamp);
```
This is enough, as it will always return a GroupVersionEntity or throw the exception.

Reply:
I'm leaving today on holiday, feel free to take hover here 🛩️